src/app/components/annotations/pages/verification/verification.component.ts

```diff
@@ -246,18 +234,107 @@ class VerificationComponent
+      console.error("Received invalid decision event", decisionEvent);
+      return;
+    }
+
+    this.hasUnsavedChanges = true;
+
+    const decision = decisionEvent.detail;
+    for (const [subject, receipt] of decision) {
+      const change = receipt.change;
+
+      if (Object.prototype.hasOwnProperty.call(change, "verification")) {
+        this.handleVerificationDecision(subject);
+      }
+
+      if (Object.prototype.hasOwnProperty.call(change, "newTag")) {
+        this.handleTagCorrectionDecision(subject);
+      }
+    }
+  }
+
+  private handleVerificationDecision(subjectWrapper: SubjectWrapper): void {
+    const subject = subjectWrapper.subject as Readonly<AudioEvent>;
+
+    // I have to use "as string" here because the upstream typing is incorrect
+    // TODO: We should remove this "as string" and improve the upstream typing
+    const mappedDecision =
+      confirmedMapping[(subjectWrapper.verification as any).confirmed];
+
+    const tagId = subjectWrapper.tag.id;
+
+    const verificationData: IVerification = {
+      audioEventId: subject.id,
+      confirmed: mappedDecision,
+      tagId,
+    };
+
+    const verification = new Verification(verificationData, this.injector);
+
+    // I have to use "as any" here to remove the readonly typing
+    const apiRequest = this.verificationApi.createOrUpdate(
+      verification,
+      subject as AudioEvent,
+      this.session.currentUser,
+    );
+
+    // I use firstValueFrom so that the observable is evaluated
+    // but I don't have to subscribe or unsubscribe.
+    // Additionally, notice that the function is not awaited so that the
+    // render thread can continue to run while the request is being made
+    firstValueFrom(apiRequest);
+  }
+
+  private handleTagCorrectionDecision(subjectWrapper: SubjectWrapper): void {
+    const audioEvent = subjectWrapper.subject as any;
+    const newTagId = subjectWrapper.newTag as any;
+
+    const apiRequest = this.correctTag(audioEvent, newTagId.tag.id);
+
+    firstValueFrom(apiRequest);
+  }
+
+  // TODO: This logic should probably be moved to a service
+  /**
+   * Corrects an incorrect tag on an audio event by verifying the existing tag
+   * as "incorrect", creating a new tag that is correct, and submitting a
+   * "correct" verification decision.
+   */
+  private correctTag(audioEvent: AudioEvent, newTagId: Id) {
+    const correctTag = new Tagging({
+      audioEventId: audioEvent.id,
+      tagId: newTagId,
+    });
+
+    return this.taggingsApi
+      .create(correctTag, audioEvent.audioRecordingId, audioEvent.id)
+      .pipe(
+        mergeMap(() => {
+          const correctVerification = new Verification({
+            audioEventId: audioEvent.id,
+            confirmed: ConfirmedStatus.Correct,
+            tagId: newTagId,
+          });
+
+          return this.verificationApi.createOrUpdate(
+            correctVerification,
+            audioEvent,
+            this.session.currentUser,
+          );
```

Comment:
this should not be needed: the verification grid should have emitted the verification already and it should be handles by `handleVerificationDecision`

Is this just double work? or is there a bug somewhere?

Reply:
this comment is wrong